scripts/india_census/primary_religion_data/preprocess.py

```diff
@@ -137,92 +212,135 @@ def stat_var_index_key(row, data_category_column=None):
+        # No need to add empty constraint to the list
+        if self._get_base_constraints(row) != "":
+            constraints_array.append(self._get_base_constraints(row))
+
+        name_array.append(title_case(data_category))
+        row["description"] = row["description"] + " - " + data_category
+        constraints_array.append("religion: " +
+                                 _RELIGION_DCID_MAPPING[data_category])
 
         if row["age"] == "YearsUpto6":
             name_array.append("YearsUpto6")
+            constraints_array.append("age: dcid:YearsUpto6")
 
         if row["socialCategory"] == "ScheduledCaste":
             name_array.append("ScheduledCaste")
+            constraints_array.append("socialCategory: dcs:ScheduledCaste")
         if row["socialCategory"] == "ScheduledTribe":
             name_array.append("ScheduledTribe")
+            constraints_array.append("socialCategory: dcs:ScheduledTribe")
 
         if row["literacyStatus"] == "Literate":
             name_array.append("Literate")
-        if row["literacyStatus"] == "Illiterate":
+            constraints_array.append("literacyStatus: dcs:Literate")
+        elif row["literacyStatus"] == "Illiterate":
             name_array.append("Illiterate")
-        else:
-            pass
+            constraints_array.append("literacyStatus: dcs:Illiterate")
 
         if row["workerStatus"] == "Worker":
             if row["workerClassification"] == "MainWorker":
                 name_array.append("MainWorker")
+                constraints_array.append("workerClassification: dcs:MainWorker")
                 if row["workCategory"] != "":
                     name_array.append(row["workCategory"])
+                    constraints_array.append("workCategory: dcs:" +
+                                             row["workCategory"])
 
             elif row["workerClassification"] == "MarginalWorker":
                 name_array.append("MarginalWorker")
+                constraints_array.append(
+                    "workerClassification: dcs:MarginalWorker")
 
                 if row["workCategory"] != "":
                     name_array.append(row["workCategory"])
+                    constraints_array.append("workCategory: dcs:" +
+                                             row["workCategory"])
 
                 if row["workPeriod"] == "[Month - 3]":
                     name_array.append("WorkedUpto3Months")
+                    constraints_array.append("workPeriod:" + row["workPeriod"])
 
                 if row["workPeriod"] == "[Month 3 6]":
                     name_array.append("Worked3To6Months")
+                    constraints_array.append("workPeriod:" + row["workPeriod"])
+
             else:
                 name_array.append("Workers")
 
         elif row["workerStatus"] == "NonWorker":
             name_array.append("NonWorker")
+            constraints_array.append("workerStatus: dcs:NonWorker")
```

Comment:
workerStatus is deprecated now and superceded by workerClassification.
https://datacommons.org/browser/workerClassification

Can use `workerClassification: dcs:NonWorker`